packages/teams-js/src/public/marketplace.ts

```diff
@@ -0,0 +1,211 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { Currency, Locale, Market } from '.';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Represents the cart object for the app checkout flow.
+ *
+ * @beta
+ */
+export interface Cart {
+  /** Version of the cart. */
+  readonly version: CartVersion;
+  /** The id of the cart. */
+  readonly id: string;
+  /** The cart info. */
+  cartInfo: CartInfo;
+  /** The cart items. */
+  cartItems: CartItem[];
+}
+
+/**
+ * Version of the cart.
+ *
+ * @beta
+ */
+interface CartVersion {
+  /** Represents the major version number. */
+  majorVersion: number;
+  /** Represents the minor version number. */
+  minorVersion: number;
+}
+
+/**
+ * Represents the cart information
+ *
+ * @beta
+ */
+interface CartInfo {
+  /** The country market where the products are selling. */
+  readonly market: Market;
+  /** The identifier to tell the cart is checked out by admin or information worker */
+  readonly intent: Intent;
+  /** Locale for the user */
+  readonly locale: Locale;
+  /** The status of the cart. */
+  status: CartStatus;
+  /** Currency code for the cart item price. */
+  readonly currency: Currency;
+  /** ISO 8601 timestamp string, indicates when the cart is created. */
+  readonly createdAt: string;
+  /** ISO 8601 timestamp string, indicates when the cart is updated. */
+  updatedAt: string;
+}
+
+/**
+ * Represents the basic cart item information.
+ *
+ * @beta
+ */
+interface Item {
+  /** The id of the cart item. */
+  readonly id: string;
+  /** The display name of the cart item. */
```

Comment:
these 3 (or at least name and price) seem like they should be readonly: I assume those values won't change after construction of the Item. `readonly` prevents changing the value of the property after construction, but you (or devs) can modify by making copies of the Item and replacing certain values at construction time.

Reply:
will make name readonly, but we allow the app to change price and quantity because the store in the host only provides the catalog service, and users are actually purchasing from the app. That means the price and quantity should respect what the app offers when the price is stale or the inventory is low. In this case, we will need the app to update the price/quantity of the host cart, through the addOrUpdateCartItems method.